src/tests/test_gfd_mining.cpp

```diff
@@ -0,0 +1,118 @@
+#include <cstdlib>
+#include <filesystem>
+#include <fstream>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/gfd/gfd_miner.h"
+#include "config/names.h"
+#include "csv_config_util.h"
+
+using namespace algos;
+using algos::StdParamsMap;
+
+namespace tests {
+
+class GfdMiningTest : public ::testing::Test {
+public:
+    std::filesystem::path const current_path = kTestDataDir / "graph_data";
+
+    static std::unique_ptr<algos::GfdMiner> CreateGfdMiningInstance(
+            std::filesystem::path const& graph_path, std::size_t const k, std::size_t const sigma) {
+        StdParamsMap option_map = {{config::names::kGraphData, graph_path},
+                                   {config::names::kGfdK, k},
+                                   {config::names::kGfdSigma, sigma}};
+        return algos::CreateAndLoadAlgorithm<GfdMiner>(option_map);
+    }
+};
+
+TEST_F(GfdMiningTest, TestMinGfd) {
+    std::filesystem::path const gfd_path = GfdMiningTest::current_path / "blogs_gfd.dot";
+    std::ifstream f(gfd_path);
+    Gfd gfd = parser::graph_parser::ReadGfd(f);
+    std::vector<Gfd> gfds = {gfd};
+
+    std::filesystem::path const graph_path = GfdMiningTest::current_path / "blogs_graph.dot";
+    auto algorithm = CreateGfdMiningInstance(graph_path, 2, 3);
```

Comment:
I don't like this listing of a set of numbers in the parameters. On the one hand, two numbers are not a big problem, but on the other hand, it's better to do it right from the start.

For this, I suggest defining a public config in the test class, which will contain these two parameters.

And then pass these configs in tests, using aggregate initialization. And readability will be much better:
```cpp
auto algorithm = CreateGfdMiningInstance(graph_path, {.k = 2, .sigma = 3});
```

If you wish, you can add the path to these parameters. There is no big difference.